usr/src/uts/common/fs/zfs/arc.c

```diff
@@ -1036,6 +1036,8 @@ uint64_t l2arc_feed_min_ms = L2ARC_FEED_MIN_MS;	/* min interval milliseconds */
 boolean_t l2arc_noprefetch = B_TRUE;		/* don't cache prefetch bufs */
 boolean_t l2arc_feed_again = B_TRUE;		/* turbo warmup */
 boolean_t l2arc_norw = B_TRUE;			/* no reads during writes */
+boolean_t arc_evict_l2_first = B_FALSE;		/* first evict buffers from ARC which are in L2ARC */
+boolean_t arc_evict_l2_only = B_FALSE;		/* only evict buffers from ARC which are in L2ARC */
```

Comment:
We can see the "drawback" of ZFS with `arc_evict_l2_only` in run 1 of my performance test :
https://www.illumos.org/issues/7361
It sometimes takes slightly more time to browse a dataset than without.

But as soon as the (meta)data is accessed again (run 2 and more), `arc_evict_l2_only` performance improvement is huge, keeping it sustained over the time.